### PR TITLE
Add `.jsonc` extension

### PR DESCRIPTION
**Pull request by nathanshelly**
Marks `.jsonc` files as `jsonc`.

## Checklist:

- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      - https://github.com/search?p=1&q=extension%3Ajsonc+NOT+dkfgjdkfjd&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s): https://github.com/neoclide/coc.nvim
    - Sample license(s): MIT

### CODE REVIEW

**Comment by nathanshelly** [comment]
This PR adds the `.jsonc` extension that was in the initial draft of #4171. It was removed per [this comment](https://github.com/github/linguist/pull/4171#discussion_r197410964). I'm adding it back hoping that usage is high enough to accept it this time. Feel free to close out immediately if it isn't.

File: samples/JSON with Comments/person.jsonc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// example comment explaining schema
```
Comment (pchaigno) [inline samples/JSON with Comments/person.jsonc]: Where is this sample from? It seems a bit too simple...

Comment (nathanshelly) [reply samples/JSON with Comments/person.jsonc]: It's a copy of the [person.json](https://github.com/github/linguist/blob/master/samples/JSON/person.json) from the plain `.json` samples with the added comment. I figured trying to add more comments might be contrived.

[Here's the file](https://github.com/nathanshelly/.files/blob/master/neovim/coc.jsonc) that prompted me opening this PR. I could add it as the sample if it's better or happy to do whatever suggestion you might have have!

Comment (pchaigno) [reply samples/JSON with Comments/person.jsonc]: > [Here's the file](https://github.com/nathanshelly/.files/blob/master/neovim/coc.jsonc) that prompted me opening this PR. I could add it as the sample if it's better or happy to do whatever suggestion you might have have!

Yes, please do. That one is much better :-)

Comment (nathanshelly) [reply samples/JSON with Comments/person.jsonc]: updated with the better sample 😄

**Comment by nathanshelly** [comment]
Could anyone explain how I would get GitHub to highlight [this `.jsonc` file](https://github.com/nathanshelly/.files/blob/master/neovim/coc.jsonc) as `jsonc`? That was my intent with opening this PR, apologies for apparently misunderstanding how GitHub's highlighting works.

I see [this section](https://github.com/github/linguist#theres-a-problem-with-the-syntax-highlighting-of-a-file) about issues with syntax highlighting but as far as I can tell the file isn't being recognized as a `jsonc` filetype so the issue happens before the upstream grammar is even applied.

**Comment by Alhadis** [comment]
@nathanshelly Changes to Linguist don't become visible on GitHub until there's a new release, which @lildude is in the process of preparing. You'll see the highlighting once #4816 goes live.

**Comment by nathanshelly** [comment]
🤦‍♂️ of course that makes perfect sense, thanks!

**Pull request merged**